pkg/option/config.go

```diff
@@ -2409,9 +2409,9 @@ func (c *DaemonConfig) NodeEncryptionEnabled() bool {
 	return c.EncryptNode
 }
 
-// EncryptionEnabled returns true if encryption is enabled
+// EncryptionEnabled returns true if IPsec or WireGuard is enabled
 func (c *DaemonConfig) EncryptionEnabled() bool {
-	return c.EnableIPSec
+	return c.EnableIPSec || c.EnableWireguard
 }
```

Comment:
I appreciate the sentiment, and agree this might even fix bugs. But it's also an easy way to land subtle regressions, and will make backports harder (now you have to evaluate the users of `EncryptionEnabled()` in stable branches).

So I'd strongly suggest to address this in a separate PR, and give @cilium/wireguard opportunity to scrutinize the change on its own.